syntax = "proto3";

package nextTwirpTest.user;
option go_package = "user";

message UserData {
  string id = 1;
  string email = 2;
  string givenName = 3;
  string familyName = 4;
  string imageUrl = 5;
}

enum IdentityProvider {
  GOOGLE = 0;
  OKTA = 1;
}

service User {
  rpc TokenExchange(TokenExchangeRequest) returns (TokenExchangeResponse);
}

message TokenExchangeRequest {
  IdentityProvider identityProvider = 1;
  string idpAccessToken = 2;
}

message TokenExchangeResponse {
  string accessToken = 1;
  UserData userData = 2;
}
